sheepdog/transactions/upload/sub_entities.py

```diff
@@ -339,37 +343,55 @@ def _update_index(self):
                     document.patch()
             return
 
-        old_doc = document.to_json()
-        document.delete()
+        is_released = document.version is not None and document.metadata.get("release_number") is not None
 
-        # Re-create URL, since the filename might have changed.
-        # NOTE: We don't care about other URLs, since the file was not yet
-        # released, so it won't have any urls other than primary url
-        urls = [
-            generate_s3_url(
-                host=self._config['SUBMISSION']['host'],
-                bucket=self._config['SUBMISSION']['bucket'],
-                program=self.transaction.program,
-                project=self.transaction.project,
-                uuid=self.entity_id,
-                file_name=self.doc['file_name']
-            )
-        ]
-        urls_metadata = {
+        # if document is not released, allow recreation of urls and url_metadata
+        if not is_released:
+            document.urls = []
+            document.urls_metadata = {}
+
+        # generate new urls if none is specified
+        if not self.urls:
```

Comment:
self.urls might not be what you think it is, you could be always None, just double check..

Reply:
The plan is whenever a url is not set re-generate the url, try to append all supplied urls to existing document if they do not already exist

Reply:
with the addition of file_name, it no longer made sense to update released files on indexd since the user submittable portion of an index doc already triggers a new version. For unreleased nodes, the existing logic is similar, just added metadata and version just to be sure the previous state of the record is captured